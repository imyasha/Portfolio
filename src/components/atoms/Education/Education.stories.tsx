import { Education, EducationProps } from './Education';

export default {
	title: 'atoms/Education',
	component: Education,
};

export const EducationStory = (args: EducationProps) => <Education {...args} />;

EducationStory.storyName = 'Education';
EducationStory.args = {
	education: {
		institute: 'Harvard X',
		course: 'CS50',
		endDate: '2020-06-20',
		place: 'remote',
		technologies: ['C', 'Python', 'SQL', 'Algorithms', 'Data Structures'],
	},
};
