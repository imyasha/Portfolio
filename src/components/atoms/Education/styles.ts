import styled from 'styled-components';

const StyledEducation = styled.div`
	margin-bottom: 3rem;
`;

const Flex = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	margin-bottom: 0.5rem;

	@media screen and (min-width: 768px) {
		flex-direction: row;
	}
`;

export { Flex, StyledEducation };
