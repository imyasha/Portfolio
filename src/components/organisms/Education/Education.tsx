import classNames from 'classnames';
import { IEducation } from '@types';

import { School } from 'Molecules/School';

export interface EducationProps {
	education: IEducation[];
}

const Education = ({ education }: EducationProps) => {
	return (
		<>
			{education.map(
				({ course, date, institute, technologies }, index: number) => (
					<div
						className={classNames(
							'flex transition-all transform md:hover:scale-[1.01]',
							{
								'justify-end': index % 2 !== 0,
							}
						)}
						key={course}
					>
						<School
							course={course}
							date={date}
							institute={institute}
							technologies={technologies}
						/>
					</div>
				)
			)}
		</>
	);
};

export { Education };
