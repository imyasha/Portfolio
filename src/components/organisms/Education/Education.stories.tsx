import { mockSchool } from 'MockData';
import { Education, EducationProps } from './Education';

export default {
	title: 'organisms/Education',
	component: Education,
};

export const EducationStory = (args: EducationProps) => <Education {...args} />;

EducationStory.storyName = 'Education';
EducationStory.args = {
	education: [mockSchool],
};
