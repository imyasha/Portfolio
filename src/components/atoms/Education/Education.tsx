import { FC } from 'react';
import { format } from 'date-fns';
import { IEducation } from '@Types';
import { Flex, StyledEducation } from './styles';

export interface EducationProps {
	education: IEducation;
}

const Education: FC<EducationProps> = ({ education }) => {
	return (
		<StyledEducation>
			<Flex>
				<div>
					<strong>
						{education.website ? (
							<a
								href={education.website}
								target="_blank"
								rel="noopener noreferrer"
							>
								{education.course}
							</a>
						) : (
							education.course
						)}
					</strong>{' '}
					- {education.institute}
				</div>
				<div>{format(new Date(education.endDate), 'MM/yyyy')}</div>
			</Flex>
			<small>
				<em>
					<strong>Course contents:</strong> {education.technologies.join(', ')}
				</em>
			</small>
		</StyledEducation>
	);
};

export { Education };
