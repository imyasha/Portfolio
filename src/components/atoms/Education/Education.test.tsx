import { Education } from '../Education';
import { render } from '@testing-library/react';

describe('Education', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Education
				education={{
					institute: 'Harvard X',
					website: 'https://cs50.harvard.edu/x/',
					course: 'CS50',
					endDate: '2020-06-20',
					place: 'remote',
					technologies: ['C', 'Python', 'SQL', 'Algorithms', 'Data Structures'],
				}}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
