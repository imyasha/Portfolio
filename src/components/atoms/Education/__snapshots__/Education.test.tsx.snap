// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education renders correctly 1`] = `
<div>
  <div
    class="styles__StyledEducation-sc-1irolb5-0 bjYQZD"
  >
    <div
      class="styles__Flex-sc-1irolb5-1 bTesre"
    >
      <div>
        <strong>
          <a
            href="https://cs50.harvard.edu/x/"
            rel="noopener noreferrer"
            target="_blank"
          >
            CS50
          </a>
        </strong>
         
        - 
        Harvard X
      </div>
      <div>
        06/2020
      </div>
    </div>
    <small>
      <em>
        <strong>
          Course contents:
        </strong>
         
        C, Python, SQL, Algorithms, Data Structures
      </em>
    </small>
  </div>
</div>
`;
